x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/PercentileMethodConfiguration.java

```diff
@@ -0,0 +1,119 @@
+
+import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isEnum;
+import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isFoldable;
+import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isInteger;
+import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isNumeric;
+
+public abstract class PercentileMethodConfiguration {
+
+    private static final Set<String> ACCEPTED_METHODS =
+        Arrays.stream(PercentilesMethod.values()).map(p -> p.getParseField().getPreferredName()).collect(Collectors.toSet());
+
+    public static Expression defaultMethod(Source source, Expression method) {
+        return method == null
+            ? new Literal(source, PercentilesMethod.TDIGEST.getParseField().getPreferredName(), DataTypes.KEYWORD)
+            : method;
+    }
+
+    public static Expression defaultMethodParameter(Expression methodParameter) {
+        return nullIfUnspecified(methodParameter);
+    }
+
+    public static Expression.TypeResolution resolvePercentileConfiguration(
+        String sourceText, Expression method, Expressions.ParamOrdinal methodOrdinal,
+        Expression methodParameter, Expressions.ParamOrdinal methodParameterOrdinal) {
+
+        Expression.TypeResolution resolution = isEnum(method, sourceText, methodOrdinal, ACCEPTED_METHODS);
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+
+        if (methodParameter != null) {
+            resolution = isFoldable(methodParameter, sourceText, methodParameterOrdinal);
+            if (resolution.unresolved()) {
+                return resolution;
+            }
+
+            PercentilesMethod percentilesMethod = asPercentilesMethod(method);
+            switch (percentilesMethod) {
+                case TDIGEST:
+                    resolution = isNumeric(methodParameter, sourceText, methodParameterOrdinal);
+                    break;
+                case HDR:
+                    resolution = isInteger(methodParameter, sourceText, methodParameterOrdinal);
+                    break;
+                default:
+                    throw new QlIllegalArgumentException("Not handled PercentilesMethod [" + percentilesMethod + "]");
+            }
+            if (resolution.unresolved()) {
+                return resolution;
+            }
+        }
+        return resolution;
+    }
+
+    private static PercentilesMethod asPercentilesMethod(Expression method) {
+        PercentilesMethod percentilesMethod = null;
+        String methodName = foldOptionalNullable(method, DataTypes.KEYWORD);
+
+        for (PercentilesMethod m : PercentilesMethod.values()) {
+            if (m.getParseField().getPreferredName().equals(methodName)) {
+                percentilesMethod = m;
+            }
+        }
+        if (percentilesMethod == null) {
+            throw new QlIllegalArgumentException("specified method [ " + methodName + " ] is not one of the allowed methods "
```

Comment:
I think here you need to not throw an exception, but create a `TypeResolution` to express the inconsistency.